bundles/org.openhab.binding.luftdateninfo/src/main/java/org/openhab/binding/luftdateninfo/internal/handler/BaseSensorHandler.java

```diff
@@ -0,0 +1,158 @@
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateFromCache();
+        }
+    }
+
+    @Override
+    public void initialize() {
+        lifecycleStatus = LC_INITIALIZING;
+        scheduler.execute(() -> {
+            config = getConfigAs(LuftdatenInfoConfiguration.class);
+            configStatus = checkConfig(config);
+            if (configStatus == CONFIG_OK) {
+                update();
+                if (updateStatus == UPDATE_OK) {
+                    updateStatus(ThingStatus.ONLINE);
+                    logger.debug("Start refresh job at interval {} min.", refreshInterval);
+                    if (refreshJob != null) {
+                        refreshJob.cancel(true);
+                    }
+                    refreshJob = scheduler.scheduleWithFixedDelay(this::update, 5, refreshInterval * 60,
+                            TimeUnit.SECONDS);
+                } else {
+                    switch (updateStatus) {
+                        case UPDATE_CONNECTION_ERROR:
+                            logger.warn("Update failed due to Connection error. Trying to recover in next refresh");
+                            break;
+                        case UPDATE_VALUE_ERROR:
+                            logger.warn(
+                                    "Sensor values doesn't match - please check if Sensor ID is delivering the correct Thing channel values");
+                            break;
+                        case UPDATE_VALUE_EMPTY:
+                            logger.warn(
+                                    "No values deliverd by Sensor. Please check for valid Sensor ID in configuration");
+                            break;
+                    }
+                    updateStatus(ThingStatus.OFFLINE);
+                }
+            } else {
+                logger.warn("Configuration not valid. Sensor ID as a number is mandatory!");
+                updateStatus(ThingStatus.OFFLINE);
+            }
+            lifecycleStatus = LC_RUNNING;
+        });
+    }
+
+    @Override
+    public void dispose() {
+        if (refreshJob != null) {
+            refreshJob.cancel(true);
+        }
+        lifecycleStatus = LC_DISPOSED;
+    }
+
+    /**
+     * Checks if config is valid - a) not null and b) sensorid is a number
+     *
+     * @param c
+     * @return
+     */
+    private int checkConfig(@Nullable LuftdatenInfoConfiguration c) {
+        if (c != null) {
+            try {
+                Integer.parseInt(c.sensorid);
+                return CONFIG_OK;
+            } catch (NumberFormatException t) {
+                return CONFIG_SENSOR_NUMBER;
+            }
+        } else
+
+        {
+            return CONFIG_IS_NULL;
+        }
+    }
+
+    public int getLifecycleStatus() {
+        return lifecycleStatus;
+    }
+
+    protected void update() {
+        String response = HTTPHandler.getResponse(config.sensorid);
```

Comment:
Config can be null at this point.

Reply:
Shouldn't be the case - update & scheduler are only started if initialize is checked with OK